lib/client.js

```diff
@@ -143,6 +155,30 @@ extend(Raven.prototype, {
     return this;
   },
 
+  makeErrorHandler: function () {
+    var self = this;
+    var calledUncaughtHandler = false;
+    var calledFatalError = false;
+    var firstError;
+    return function (err) {
+      if (!calledUncaughtHandler) {
+        firstError = err;
+        calledUncaughtHandler = true;
+        self.captureException(err, function (sendErr, eventId) {
+          calledFatalError = true;
+          self.onFatalError(err, sendErr, eventId);
+        });
+      } else if (!calledFatalError) {
+        self.onFatalError(firstError, err);
+      } else {
+        // we hit an error *after* calling onFatalError - pretty fucked
```

Comment:
it's prrrrrobably better not to say 'fucked' -- it won't be in the minified code but I could imagine some kind of automatic censorship causing a problem or something. idk